niworkflows/interfaces/nibabel.py

```diff
@@ -13,6 +13,45 @@
 IFLOGGER = logging.getLogger('nipype.interface')
 
 
+class _ApplyMaskInputSpec(BaseInterfaceInputSpec):
+    in_file = File(exists=True, mandatory=True, desc='an image')
+    in_mask = File(exists=True, mandatory=True, desc='a mask')
+    threshold = traits.Float(0.5, usedefault=True,
+                             desc='a threshold to the mask, if it is nonbinary')
+
+
+class _ApplyMaskOutputSpec(TraitedSpec):
+    out_file = File(exists=True, desc='masked file')
+
+
+class ApplyMask(SimpleInterface):
+    """Mask the input given a mask."""
+
+    input_spec = _ApplyMaskInputSpec
+    output_spec = _ApplyMaskOutputSpec
+
+    def _run_interface(self, runtime):
+        img = nb.load(self.inputs.in_file)
+        msknii = nb.load(self.inputs.in_mask)
+        msk = msknii.get_fdata() > self.inputs.threshold
+
+        self._results['out_file'] = fname_presuffix(
+            self.inputs.in_file, suffix='_masked', newpath=runtime.cwd)
+
+        if img.dataobj.shape[:3] != msk.shape:
+            raise ValueError("Image and mask sizes do not match.")
+
+        if not np.allclose(img.affine, msknii.affine):
+            raise ValueError("Image and mask affines are not similar enough.")
+
+        if img.dataobj.ndim == msk.ndim + 1:
+            msk = msk[..., np.newaxis]
+
+        masked = img.__class__(img.dataobj * msk, None, img.header)
```

Comment:
This is probably the potentially contended bit of this new interface. I'll leave it here and won't merge until it gets the blessings from @effigies, whenever he's back and can have a look.

Reply:
Seems fine. `numpy.ndarray` defines `__rmul__`, so this will act like `np.asarray(img.dataobj) * mask`.